src/pages/index.js

```diff
@@ -49,10 +99,53 @@ constants.addNewCardButton.addEventListener("click", () => {
 });
 
 function handleImageSubmit(card) {
-  renderCard(card);
-  addNewCardModal.close();
-  constants.imageSubmitForm.reset();
-  addImageFormValidator.resetValidation();
+  addNewCardModal.setLoading(true, "Saving...");
+  api
+    .addCard(card)
+    .then(() => {
+      renderCard(card);
+    })
+    .finally(() => {
+      addNewCardModal.setLoading(false);
+      constants.imageSubmitForm.reset();
+      addImageFormValidator.resetValidation();
+      addNewCardModal.close();
+    });
+}
+
+//Card like && dislike
+
+function likeButton(id) {
+  api.getInitialCards().then((data) =>
```

Comment:
Needs correction: Here you should make add or remove likes request. Please don't get the initial cards.